python/ray/data/tests/test_stats.py

```diff
@@ -147,16 +147,18 @@ def enable_get_object_locations_flag():
 @contextmanager
```

Comment:
can you also add a unit test for the `StatsManager` class? There are a few things worth testing:
* Multiple datasets/iterators running in different threads. 
* The thread can restart after stopping.